engine/class_modules/priest/sc_priest.cpp

```diff
@@ -2034,12 +2033,24 @@ void priest_t::arise()
   buffs.whispers_of_the_damned->trigger();
 }
 
+void priest_t::trigger_shadowflame_prism( player_t* target )
```

Comment:
Some things which should probably still be improved here:
1. Check if current_pet is nullptr or not. Either add an assert, or if it is possible to trigger shadowflame prism without having a main pet (eg. no summon lines in the APL) just return without doing anything.

2. Move the assert current_pet->shadowflame_prism inside the !sleeping condition.

3. Why did you remove the set_target line? I would assume it works again after your other fixes.